recipes/minisat/all/conanfile.py

```diff
@@ -0,0 +1,85 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, collect_libs, copy, export_conandata_patches, get
+from conan.tools.microsoft import is_msvc
+from os.path import join
+
+required_conan_version = ">=1.53.0"
+
+
+class MiniSatConan(ConanFile):
+    name = "minisat"
+    description = "minimalistic SAT solver"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://minisat.se"
+    topics = ("satisfiability", "solver")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("zlib/1.2.13", transitive_headers=True)
```

Comment:
Quick question, can you show me where `transitive_headers` would be needed for this library please? I looked around in its repo but couldn't see anything and I fear I might be missing something :)

Reply:
in the test package (and any other place where minisat would be used):

including core/Solver.h
includes minisat/utils/Options.h
which includes minisat/utils/ParseUtils.h
which includes zlib

thus, the public headers of minisat require the zlib header

Reply:
Oh it's in `minisat/utils/ParseUtils.h`, didn't see it, thanks for the clarification! :)

Reply:
```suggestion
        # https://github.com/niklasso/minisat/blob/37dc6c67e2af26379d88ce349eb9c4c6160e8543/minisat/utils/ParseUtils.h#L27
        self.requires("zlib/1.2.13", transitive_headers=True)
```

Reply:
and `transitive_libs` is also required obviously: https://github.com/niklasso/minisat/blob/37dc6c67e2af26379d88ce349eb9c4c6160e8543/minisat/utils/ParseUtils.h#L49

I guess it doesn't fail in test package because `assureLookahead()` member function of `StreamBuffer` class in not called.